action.yml

```diff
@@ -0,0 +1,103 @@
+---
+name: 'Docker-build-release'
+description: 'Build Docker image and push to GHCR'
+inputs:
+  organization:
+    description: 'Organizational host for the image. Defaults to the organization of the calling repository.'
+  metadata-file:
+    description: 'Metadata YAML file containing information'
+    default: metadata.yaml
+  image-name-key-path:
+    description: 'YAML path to key for image name'
+    default: '.image_name'
+  github-token:
+    description: 'GITHUB_TOKEN for access'
+    default: ${{ github.token }}
+  custom-tags:
+    description: 'Custom tags'
+    default: ''
+  context:
+    description: 'Docker build context'
+    default: '.'
+  non-semver-tags:
+    description: 'If set to any value, build all tags beginning with v'
+    default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Read YAML
+      id: yaml-data
+      uses: mikefarah/yq@v4.44.2
+      with:
+        cmd: yq '${{ inputs.image-name-key-path }}' '${{ inputs.metadata-file }}'
```

Comment:
**Question**: Is it possible for arbitrary commands to be executed here, or perhaps possible to read arbitrary files? For example, if `inputs.image-name-key-path` is `'; echo $GITHUB_TOKEN ;'`

Now I'm wondering about this in some other actions we have ...

Reply:
Excellent catch - GitHub [recommends](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable) using intermediate environment variables for this.

Reply:
Thanks - I vaguely remember we talked about this somewhere else.

Reply:
Argh, this isn't an easy fix - the whole point of the intermediate envvars is to _not_ use the inputs to construct a shell command, but the `yq` action only supports passing in a string to be `eval`d. I'll have to look at this next week to work something else out.

Reply:
Sounds good! A suggestion may be to escape any single quotes, since that is (afaik) the only way to escape the outer single quotes in this case.

Reply:
(╯°□°)╯︵ ┻━┻

I just ripped out yq entirely and used python with 57dc72d78ab2f6c0ad345d4b79298e4ba921d8bd. No shell expansion, lots of sanity-checking, and I confirmed that it works: https://github.com/uclahs-cds/docker-internal-tests/actions/runs/10427550837

Reply:
I marked this PR as closing #15 as well.

Reply:
That's one way to do it. :)